src/managers/KeybindManager.cpp

```diff
@@ -454,11 +454,33 @@ void CKeybindManager::changeworkspace(std::string args) {
     int workspaceToChangeTo = 0;
     std::string workspaceName = "";
 
+    // Flag needed so that the previous workspace is not recorded when switching
+    // to a previous workspace.
+    bool isSwitchingToPrevious = false;
+
     if (args.find("[internal]") == 0) {
         workspaceToChangeTo = std::stoi(args.substr(10));
         const auto PWORKSPACE = g_pCompositor->getWorkspaceByID(workspaceToChangeTo);
         if (PWORKSPACE)
             workspaceName = PWORKSPACE->m_szName;
+    } else if (args.find("previous") == 0) {
+        const auto P_CURRENT_WORKSPACE = g_pCompositor->getWorkspaceByID(
+            g_pCompositor->getMonitorFromCursor()->activeWorkspace);
+
+        // Do nothing if there's no previous workspace, otherwise switch to it.
+        if (P_CURRENT_WORKSPACE->m_iPrevWorkspaceID == -1) {
+            Debug::log(LOG, "No previous workspace to change to");
+            return;
+        }
+        else {
+            workspaceToChangeTo = P_CURRENT_WORKSPACE->m_iPrevWorkspaceID;
+            isSwitchingToPrevious = true;
+
+            // If the previous workspace ID isn't reset, cycles can form when continually going
+            // to the previous workspace again and again.
+            if (!g_pConfigManager->getConfigValuePtr("general:allow_workspace_cycles")->intValue)
```

Comment:
no, defeats the purpose of getConfigValuePtr.

```
static auto *const PALLOWWORKSPACECYCLES = &g_pConfigManager->getConfigValuePtr("general:allow_workspace_cycles")->intValue;
```
on top of the func, then `*PALLOWWORKSPACECYCLES`

Reply:
Will do.